packages/flutter_tools/lib/src/cache.dart

```diff
@@ -446,6 +446,29 @@ class Cache {
     return overrideUrl;
   }
 
+  String get cipdBaseUrl {
+    final String? overrideUrl = _platform.environment['FLUTTER_STORAGE_BASE_URL'];
+    if (overrideUrl == null) {
+      return 'https://chrome-infra-packages.appspot.com/dl';
```

Comment:
We already defined a hidden CIPD base URL here:
https://github.com/flutter/flutter/blob/6c1455a23ae53ec3a92be8f50a798d6c789b471f/packages/flutter_tools/lib/src/flutter_cache.dart#L578

Should we hard-code this again?

Reply:
I am removing this constant in this PR so that people do not accidentally use it in the future. Of course, the bar to make this mistake is still not very high (anyone can just create a new constant), but at least making it a local constant doesn't make it easily accessible. Actually, now that I think of it, I should probably leave a comment warning people about this.

Reply:
So it seems we should have a policy about this in the future? URLs can be easily introduced, but the mistake can be avoided most time with guidance.

Reply:
I'm not sure how to make a policy such that it's effective. A policy is just a document somewhere that someone has to read, preferably before they start making changes to the code. Maybe such document exists already, and I didn't read it. I would be more likely to read a comment in the code, although that's still not bullet proof.

I'll delegate to @christopherfujino, who is the tech lead for the tool, to decide if a policy can help here.

Reply:
Agreed with a lower priority, can other parts of this PR push forward? They are looking good so far.

Reply:
I'll update a doc about it, but agreed, the "policy" can only be enforced in code review. We did not have a policy banning this before, because we had not considered the effect on users in China. AFAIK, the only previous usage of this CIPD hack was for Fuchsia-related assets, which I'm guessing did not have many/any Chinese users.